### PR TITLE
Add Network Commissioning cluster definition

### PR DESCRIPTION
**Pull request by mfucci**

### CODE REVIEW

**Comment by mfucci** [comment]
Required for iOs support

File: src/matter/cluster/NetworkCommissioningCluster.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    attributes: {
+        maxNetworks: Attribute(0, TlvUInt8), /* read = admin */
+        networks: Attribute(1, TlvArray(TlvObject({
+            networkId: TlvField(0, TlvByteString.bound({ length: 32 })),
+            connected: TlvField(1, TlvBoolean),
+        })), { default: [] }), /* read = admin */
+        scanMaxTimeSeconds: Attribute(2, TlvUInt8),
+        connectMaxTimeSeconds: Attribute(3, TlvUInt8),
+        interfaceEnabled: WritableAttribute(4, TlvBoolean), /* write = admin */
```
Comment (Apollon77) [inline src/matter/cluster/NetworkCommissioningCluster.ts:44]: default: true

File: src/matter/cluster/NetworkCommissioningCluster.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        lastNetworkingStatus: Attribute(5, TlvEnum<NetworkCommissioningStatus>()), /* read = admin */
```
Comment (Apollon77) [inline src/matter/cluster/NetworkCommissioningCluster.ts:45]: default null, so nullable

File: src/matter/cluster/NetworkCommissioningCluster.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        lastNetworkId: Attribute(6, TlvByteString.bound({ length: 32 })), /* read = admin */
```
Comment (Apollon77) [inline src/matter/cluster/NetworkCommissioningCluster.ts:46]: nullable, default null

**Review by Apollon77 — APPROVED** [review]
New year, new review :-))

Happy new year @mfucci and have a great time coming over into "2023 that matters" ;-)

I think to get iOS Working the PR is completely fine.
Should I send in a PR to here next days to finalize the cluster specs - same for Admin Communissioning) ? Or should I send it in into matter.js?

File: src/matter/cluster/NetworkCommissioningCluster.ts (hunk 1 above)
Comment (mfucci) [reply src/matter/cluster/NetworkCommissioningCluster.ts:44]: Done

File: src/matter/cluster/NetworkCommissioningCluster.ts (hunk 2 above)
Comment (mfucci) [reply src/matter/cluster/NetworkCommissioningCluster.ts:45]: Done

**Comment by mfucci** [comment]
Yes, it would be good if you can clean up those new cluster definitions like you did for the other ones!

On my side, I am focusing on resolving all the problem for iOs / Android / Alexa support and just writing the minimal code to get stuff working.

We will move the new clusters to matter.js when there are clean.

**Comment by Apollon77** [comment]
Ok, I will start with cluster enhancements that tomorrow

**Pull request merged**